lib/alphanumeric_sorter.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+module AlphanumericSorter
+  class << self
+    def sort(values, other_last: false)
+      values.sort_by.with_index do |value, idx|
+        [
+          other_last && value.to_s.downcase == "other" ? 1 :  0,
+          *normalize_value(value),
+          idx
+        ]
+      end
+    end
+
+    def normalize_value(value)
+      @normalized_values ||= {}
+      @normalized_values[value] ||= begin
+        if (numerical = value.match(RegexPattern::NUMERIC_PATTERN))
```

Comment:
This is an anti-pattern and I declared variables above instead to avoid it. Also ends up a bit more readable. I'm surprised RuboCop isn't complaining about this, because it usually does

Reply:
I'd say it is more about the complexity of the call. Rubocop specifically complains when you don't have the `()` (which is the soft signal assignment is happening).

I opted for this pattern here to save a regex check :grin:

Reply:
If you want to keep it we can keep it. GPT suggested the same thing to me when I was writing the initial sorter. 

Anti-pattern aside I liked the readability of `if numerical do, elsif sequential do, else`